packages/web/components/Dashboard/MyFeed/MyFeed.tsx

```diff
@@ -115,130 +68,43 @@ const MyFeed: React.FC<Props> = ({ currentUser, initialSearchFilters }) => {
     router.push({ ...router, query: newQuery })
   }
 
-  const onSearchChange = useCallback((val): void => setSearchState(val), [])
-
-  const onTopicAdd = useCallback(
-    (id: number): void => {
-      setSelectedTopicsFilters([...selectedTopicsFilters, id])
-      resetPagination()
-    },
-    [selectedTopicsFilters],
-  )
-
-  const onTopicRemove = useCallback(
-    (id: number): void => {
-      setSelectedTopicsFilters(selectedTopicsFilters.filter((topicId) => topicId !== id))
-      resetPagination()
-    },
-    [selectedTopicsFilters],
-  )
-
-  const onLanguageAdd = useCallback(
-    (id: number): void => {
-      setSelectedLanguageFilters([...selectedLanguageFilters, id])
-      resetPagination()
-    },
-    [selectedLanguageFilters],
-  )
-
-  const onLanguageRemove = useCallback(
-    (id: number): void => {
-      setSelectedLanguageFilters(selectedLanguageFilters.filter((languageId) => languageId !== id))
-      resetPagination()
-    },
-    [selectedLanguageFilters],
-  )
-
   useEffect(() => {
     const searchFilters = {
-      languages: selectedLanguageFilters,
-      topics: selectedTopicsFilters,
-      needsFeedback: needsFeedbackFilter,
-      hasInteracted: hasInteractedFilter,
+      languages: postQueryVars?.languages,
+      topics: postQueryVars?.topics,
+      needsFeedback: postQueryVars?.needsFeedback,
+      hasInteracted: postQueryVars?.hasInteracted,
     }
     document.cookie = `default_search_filters=${JSON.stringify(searchFilters)};`
-  }, [selectedLanguageFilters, selectedTopicsFilters, needsFeedbackFilter, hasInteractedFilter])
+  }, [
+    postQueryVars?.languages,
+    postQueryVars?.topics,
+    postQueryVars?.needsFeedback,
+    postQueryVars?.hasInteracted,
```

Comment:
Same here, although since these values are now rolled together in `postQueryVars` and we never mutate that in place (i.e. we always create a new object when calling `setPostQueryVars`) this dependency array can just be `[postQueryVars]`. We'll run this `useEffect` slightly more often than necessary, like when the search value changes, but writing to a cookie is cheap.

Reply:
Donito!